server_modules/file_system.js

```diff
@@ -51,13 +49,15 @@ exports.readFile = function(file) {
     });
 };
 
-exports.writeFile = function(file, contents, cb) {
+exports.writeFile = function(file, contents) {
     file = exports.resolve(file);
 
-    var dir = path.dirname(file);
-    mkdirp(dir, function(err) {
-        if (err) return cb(err);
-        fs.writeFile(file, contents, cb);
+    return Q.promise(function(resolve, reject) {
```

Comment:
Promises are native to node and browsers these days, I think we can safely remove Q by now